packager/app/packager_main.cc

```diff
@@ -415,6 +416,14 @@ Status RunRemuxJobs(const std::vector<std::unique_ptr<RemuxJob>>& remux_jobs) {
 }
 
 bool RunPackager(const StreamDescriptorList& stream_descriptors) {
+#if defined(OS_WIN)
+  WSADATA wsa_data;
+  if (WSAStartup(MAKEWORD(2,2), &wsa_data)) {
+    LOG(ERROR) << "Winsock start up failed.";
+    return false;
+  }
+#endif  // defined(OS_WIN)
```

Comment:
Prefer calling WSAStartup only if it is needed. We can use static initializer in upd_file implementation to do that.

Here is one possible implementation:

```cpp
class LibWinsockInitializer {
 public:
  LibWinsockInitializer() {
    WSADATA wsa_data;
    error_ = WSAStartup(MAKEWORD(2, 2), &wsa_data);
  }

 ~LibWinsockInitializer() {
  if (error_ == 0) WSACleanup();
 }

 int error() const { return error_; }

 private:
  int error_;

  DISALLOW_COPY_AND_REASSIGN(LibWinsockInitializer);
};

bool UdpFile::Open() {
  static LibWinsockInitializer lib_winsock_initializer;
  if (lib_winsock_initializer.error() != 0) {
    LOG(ERROR) << "Winsock start up failed with error " << lib_winsock_initializer.error();
    return false;
  }
  ...
}

```